src/typescript/sdk/src/tx-builder.ts

```diff
@@ -0,0 +1,345 @@
+import { type BCS, type Types } from "aptos";
+import { type MaybeHexString } from "aptos";
+
+import { type AdvanceStyle, type Side } from "../types/order";
+import { type Restriction } from "../types/order";
+import { type SelfMatchBehavior } from "../types/order";
+import { type EconiaClient } from ".";
+
+export class EconiaTransactionBuilder {
+  client: EconiaClient;
+  entry?: Types.EntryFunctionPayload;
```

Comment:
Why have `entry` be an optional field? Why not just have all of the functions below return a `Types.EntryFunctionPayload`?